Kernel/Syscalls/fallocate.cpp

```diff
@@ -37,7 +37,8 @@ ErrorOr<FlatPtr> Process::sys$posix_fallocate(int fd, Userspace<off_t const*> us
     if (description->is_fifo())
         return ESPIPE;
 
-    if (!S_ISREG(TRY(description->file().stat()).st_mode))
+    // [ENODEV] The fd argument does not refer to a regular file.
+    if (!description->file().is_inode())
```

Comment:
Wouldn't you still need to check `S_ISREG`? Since directories and symlinks are also inodes, but not regular files.

Reply:
Oh my goodness you are of course right. :^)

Reply:
The corresponding discord conversation is:
<https://discord.com/channels/830522505605283862/830525093905170492/987771631030141018>
(`git blame me`)

Reply:
Added `File::is_regular_file()` helper to make this nice & expressive.